src/engine/controls/bpmcontrol.cpp

```diff
@@ -944,57 +958,54 @@ void BpmControl::trackLoaded(TrackPointer pNewTrack) {
 void BpmControl::trackBeatsUpdated(mixxx::BeatsPointer pBeats) {
     if (kLogger.traceEnabled()) {
         kLogger.trace() << getGroup() << "BpmControl::trackBeatsUpdated"
-                        << (pBeats ? pBeats->getBpm() : 0.0);
+                        << (pBeats ? pBeats->getBpm() : mixxx::Bpm(0.0));
     }
     m_pBeats = pBeats;
     updateLocalBpm();
     resetSyncAdjustment();
 }
 
 void BpmControl::slotBeatsTranslate(double v) {
-    mixxx::BeatsPointer pBeats = m_pBeats;
-    if (v > 0 && pBeats && (pBeats->getCapabilities() & mixxx::Beats::BEATSCAP_TRANSLATE)) {
-        double currentSample = getSampleOfTrack().current;
-        double closestBeat = pBeats->findClosestBeat(currentSample);
-        int delta = currentSample - closestBeat;
+    if (v > 0 && m_pBeats) {
+        mixxx::Frame currentFrame(getSampleOfTrack().current / 2.0);
+        mixxx::Frame closestBeat = m_pBeats->findClosestBeat(currentFrame);
+        int delta = currentFrame.getValue() - closestBeat.getValue();
         if (delta % 2 != 0) {
```

Comment:
Samples are restricted to even numbers to land on frame boundaries. But now delta is calculated in frames and this should be obsolete. Please recheck all occurrences carefully!